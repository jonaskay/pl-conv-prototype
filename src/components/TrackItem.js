import React from 'react';

class TrackItem extends React.Component {
  render() {
    if (!this.props.resource_id) {
      return null;
    }

    const iframe = (() => {
      switch (this.props.service) {
        case 'spotify':
          return (
            <iframe
              src={'https://open.spotify.com/embed?uri=spotify:track:' + this.props.resource_id}
              width="288"
              height="80"
              frameBorder="0"
              title="Spotify Play Button"></iframe>
          );
        case 'youtube':
          return (
            <iframe
              src={'https://www.youtube.com/embed/' + this.props.resource_id + '?controls=0&amp;showinfo=0'}
              width="288"
              height="162"
              frameBorder="0"
              allow="autoplay; encrypted-media"
              allowFullScreen
               title="YouTube Player"></iframe>
          );
        default: return <div></div>;
      }
    })();

    return (
      <div style={{marginBottom: '1rem'}}>{iframe}</div>
    );
  }
}

export default TrackItem;
