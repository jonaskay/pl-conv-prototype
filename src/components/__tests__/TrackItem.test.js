import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import TrackItem from '../TrackItem';

it('renders a Spotify player', () => {
  const component = ReactTestRenderer.create(<TrackItem service="spotify" />);
  expect(component).toMatchSnapshot();
});

it('renders a YouTube player', () => {
  const component = ReactTestRenderer.create(<TrackItem service="youtube" />);
  expect(component).toMatchSnapshot();
});
