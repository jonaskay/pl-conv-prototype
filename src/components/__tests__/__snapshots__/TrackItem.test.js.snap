// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a Spotify player 1`] = `
<div
  style={
    Object {
      "marginBottom": "1rem",
    }
  }
>
  <iframe
    frameborder="0"
    height="80"
    src="https://open.spotify.com/embed?uri=spotify:track:undefined"
    title="Spotify Play Button"
    width="288"
  />
</div>
`;

exports[`renders a YouTube player 1`] = `
<div
  style={
    Object {
      "marginBottom": "1rem",
    }
  }
>
  <iframe
    allow="autoplay; encrypted-media"
    allowFullScreen={true}
    frameborder="0"
    height="162"
    src="https://www.youtube.com/embed/undefined?controls=0&amp;showinfo=0"
    title="YouTube Player"
    width="288"
  />
</div>
`;
